### PR TITLE
feat: start work on party registry

### PR DESCRIPTION
**Pull request by Alxandr**
Creates the database and data access layer for RRR.

See: #253

### CODE REVIEW

File: src/Altinn.ResourceRegistry.Core/PartyRegistry/IPartyRegistryRepository.cs [hunk 1]
```diff
@@ -0,0 +1,342 @@
+﻿using System.Collections.Immutable;
+
+namespace Altinn.ResourceRegistry.Core.PartyRegistry;
+
+/// <summary>
+/// Repository for managing party registries.
+/// </summary>
+public interface IPartyRegistryRepository
```
Comment (TheTechArch) [inline src/Altinn.ResourceRegistry.Core/PartyRegistry/IPartyRegistryRepository.cs:8]: I am possibly blind, but I belive we need to get party list based on resource/owner without the list id.

**Review by TheTechArch — APPROVED** [review]
Looking forward to see how this works in practice 💯 
New paradigm for Altinn

File: src/Altinn.ResourceRegistry.Core/PartyRegistry/IPartyRegistryRepository.cs (hunk 1 above)
Comment (Alxandr) [reply src/Altinn.ResourceRegistry.Core/PartyRegistry/IPartyRegistryRepository.cs:8]: You're not blind. It's just not part of this first PR, cause the API isn't made yet.

File: src/Altinn.ResourceRegistry.Persistence/Migration/v0.02-party-registry/01-setup-tables.sql [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+DROP TABLE IF EXISTS resourceregistry.party_registry_events;
+
+DROP TABLE IF EXISTS resourceregistry.party_registry_state;
+
+DROP TABLE IF EXISTS resourceregistry.party_registry_members_state;
+
+DROP TABLE IF EXISTS resourceregistry.party_registry_resource_connections_state;
+
+DROP TYPE IF EXISTS resourceregistry.party_registry_event_kind;
```
[the remark this answers is not in the document]
Comment (Alxandr) [reply src/Altinn.ResourceRegistry.Persistence/Migration/v0.02-party-registry/01-setup-tables.sql]: I've replaced the DROPs with IF EXISTS on the create statements instead. Should be less likely to nuke data.

**Pull request merged**